VideoLocker/src/main/java/org/edx/mobile/services/CourseManager.java

```diff
@@ -90,17 +92,20 @@ public static IBlock normalizeCourseStructure(CourseStructureV1Model courseStruc
 
     private static void normalizeCourseStructure(CourseStructureV1Model courseStructureV1Model,
                                                 BlockModel block,
-                                                CourseComponent parent){
+                                                CourseComponent parent) {
 
-        if ( block.isContainer() ){
+        if (block.isContainer()) {
             CourseComponent child = new CourseComponent(block, parent);
             for (BlockModel m : courseStructureV1Model.getDescendants(block)) {
                 normalizeCourseStructure(courseStructureV1Model, m, child);
             }
         } else {
-            if ( BlockType.VIDEO == block.type && block.data != null
-                && (block.data  instanceof VideoData) ){
+            if (BlockType.VIDEO == block.type && block.data != null
+                    && (block.data instanceof VideoData)) {
                 new VideoBlockModel(block, parent);
+            } else if (BlockType.DISCUSSION == block.type && block.data != null
+                    && (block.data instanceof DiscussionData)) {
```

Comment:
Note that the `instanceof` check performs a safe `null` check implicitly, and as such the `null` check on the `block.data` property can be removed from these `if` conditions.